boundary_bot/code_matcher.py

```diff
@@ -5,37 +5,26 @@
 
 # (fuzzy-)match string local auth names to gov.uk register codes
 class CodeMatcher:
-
     def __init__(self):
         councils = self.get_data()
-        councils = [c for c in councils if not c['end-date']]
```

Comment:
Obviously there is an extent to which attempting to assign codes to boundary reviews by fuzzy matching the local auth names is going to have some room for error and does require a manual sanity check rather than just blindly accepting it. However, there's something important to point out here.

With the previous code/data, we were filtering to only the current 'generation' of local authorities here.

One of the big problems with using https://raw.githubusercontent.com/mysociety/uk_local_authority_names_and_codes/main/data/lookup_name_to_registry.csv is that it mashes current and historic data into a flat structure with no way to filter it. This is particularly problematic when names get re-used across structural changes.

The most obviously bad case of this is Dorset. If you've got a boundary review for Dorset, the previous code (consuming https://raw.githubusercontent.com/DemocracyClub/GDSRegisters/master/registers/local-authority-eng/local-authority-eng.csv ) is going to get rid of `DOR` here and leave you with only `DST` available to be matched.

The new code here has something like a 50% chance of matching that to `DOR` (wrong) and a 50% chance of matching that to `DST` (correct) because the exact same string can exist in that file mapped to multiple different codes and you can't filter to only current authorities.

This can only get worse over time.

Reply:
Good point Chris. The reasons for the switch were not duplicating the work of maintaining the register, and the format they produced having a field containing a set of alias (which I'm not using in this PR). I guess the first is maybe better taken care of by scraping mySoc's repo, and applying our own filters. Or inspect their `current-authority` flag. Anyway options are there, I'll have a think. Thanks for the heads up!